src/librustc_save_analysis/dump_visitor.rs

```diff
@@ -115,15 +117,18 @@ impl<'l, 'tcx: 'l, 'll, D: Dump + 'll> DumpVisitor<'l, 'tcx, 'll, D> {
         where F: FnOnce(&mut DumpVisitor<'l, 'tcx, 'll, D>)
     {
         let item_def_id = self.tcx.hir.local_def_id(item_id);
+        let old_item_cx = mem::replace(&mut self.save_ctxt.item_cx,
+                                       ItemCtxt::new(self.tcx, item_def_id));
```

Comment:
Yeah, I was debating about stripping it out too -- I was thinking I could just make a `pub fn` in `librustc_typeck` that takes a def-id and constructs a "fresh itemctxt". I think when I wrote this patch I was anticipating having a cache per `ItemCtxt` (but then I checked and found that said cache had zero hits).